ptn/buttonrolebot/ui_elements/ButtonConfig.py

```diff
@@ -485,23 +503,35 @@ def __init__(self, button_data: RoleButtonData):
         self.add_item(PrevButton(self.index, self.button_data))
         self.add_item(CancelButton(self.index))
         self.add_item(self.final_submit_button)
+        self.add_item(self.add_another_button)
 
     @discord.ui.button(
-            label='✔ Create Button',
-            style=discord.ButtonStyle.success,
-            custom_id="final_submit_button",
-            row = 1
-            )
-
+        label='✔ Create Button',
+        style=discord.ButtonStyle.success,
+        custom_id="final_submit_button",
+        row=1
+    )
     async def final_submit_button(self, interaction: discord.Interaction, button):
         print("Received ✅ final_submit_button click")
         try:
             # create the button!
-            view = await _add_role_button_to_view(interaction, self.button_data)
+            button = _create_button(self.button_data)
+            self.button_data.button_list.append(button)
```

Comment:
I wonder whether it would be better to add the first button, then ask whether we want to add more, rather than storing the buttons up to add all at the same time. The Confirm interface isn't presently set up for confirming details of more than one button, and presently buttons can't be edited either. (it's also simpler, though you've already done all the coding :D )

Reply:
I ran into is the view, it seemed to overwrite the button unless I added them together

Reply:
Intersting. The code that checks whether the message has a view wasn't designed with expediency in mind, it's possible that the bot adds a button then checks the view status before it's had a chance for Discord to register the change. This seems an elegant enough way around that.